pkg/cmd/server/api/helpers.go

```diff
@@ -317,6 +331,62 @@ func GetAPIClientCertCAPool(options MasterConfig) (*x509.CertPool, error) {
 	return cmdutil.CertPoolFromFile(options.ServingInfo.ClientCA)
 }
 
+// GetNamedCertificateMap returns a map of strings to *tls.Certificate, suitable for use in tls.Config#NamedCertificates
+// Returns an error if any of the certs cannot be loaded
+// Returns nil if len(namedCertificates) == 0
+func GetNamedCertificateMap(namedCertificates []NamedCertificate) (map[string]*tls.Certificate, error) {
+	if len(namedCertificates) == 0 {
+		return nil, nil
+	}
+	namedCerts := map[string]*tls.Certificate{}
+	for _, namedCertificate := range namedCertificates {
+		cert, err := tls.LoadX509KeyPair(namedCertificate.CertFile, namedCertificate.KeyFile)
```

Comment:
> no, it'll let you use a cert to serve anything you want (it'll still secure the connection, the client just won't be able to validate the hostname). I can add validation and warn if the cert is invalid or doesn't contain a DNS name matching the configured names

Yeah, I see no valid reason for choosing a *different* cert that won't validate.  I would do it here, not in the validate methods, since we haven't been loading certs there.